### PR TITLE
ZF-55

### PR DESCRIPTION
**Pull request by wafschneider**
If restrictToItem is set, do not return a MARC holdings field for holdings records with no items.

### CODE REVIEW

File: lib/Net/Z3950/FOLIO/MARCHoldings.pm [hunk 1]
```diff
@@ -10,16 +10,21 @@ sub insertMARCHoldings {
     my $holdingsObjects = $rec->holdings($marc);
 
     for (my $i = 0; $i < @$holdingsObjects; $i++) {
-	my $holdingsMap = _listOfPairs2map($holdingsObjects->[$i]);
-
-	# Annoyingly, a field can't be created with no subfields, hence this "accumulator" approach
-	my $marcField = _addSubfields(undef, $marcCfg, $marcCfg->{holdingsElements}, $holdingsMap);
-
+        my $holdingsMap = _listOfPairs2map($holdingsObjects->[$i]);
 	my $itemObjects = $holdingsMap->{circulations};
```
[the remark this answers is not in the document]
Comment (MikeTaylor) [reply lib/Net/Z3950/FOLIO/MARCHoldings.pm:14]: After a looong process of replicating this (on a box where I had not previously built this software, and in an adventure that involved creating https://github.com/MikeTaylor/yaz-config-polyfill ), I am finally able to verify that this not only looks right, it works right. Thank you for putting this together.

**Review by MikeTaylor — CHANGES_REQUESTED** [review]
Apart from fixing the space/tab issue, the only other thing we need here is a `Changes.md` entry.

File: lib/Net/Z3950/FOLIO/MARCHoldings.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (wafschneider) [reply lib/Net/Z3950/FOLIO/MARCHoldings.pm:14]: I've pushed an updated file with the indentation corrected, and added an `.editorconfig` file to at least document the convention. That is kind of a weird one, sorry I was unaware (`M-x indent-region` with the indent set to 4 and style set to space was the culprit for my initial commit). Thanks for reviewing, let me know if I got it right.

[the remark this answers is not in the document]
Comment (MikeTaylor) [reply lib/Net/Z3950/FOLIO/MARCHoldings.pm:14]: I don't blame you at all, I admit it is an objectively bad convention.

**Review by MikeTaylor — APPROVED** [review]
Looking good! Thank you for handling this.

**Comment by dcrossleyau** [comment]
(The PR comments above lost their order flow, so continuing here.)

@MikeTaylor: Regarding Markdown line-breaks:

Oh, yeah i had not considered PDF.

Did some investigation today and found that another way to cause a hard line-break in documents generated from Markdown is to terminate the line with a backslash `\` e.g.\
rather than using the html `<br />` element.

See [example](https://github.com/folio-org/folio-org.github.io/commit/48295900c3adb42efca7cce368e2e62c2620ef04) in Markdown, and presented via the https://dev.folio.org/reference/api/#versioned-api-docs

My 'pandoc' is a bit old to attempt a proof demonstration, but i did find some useful resources. (I know that not all Markdown parsers support these methods.)

* https://stackoverflow.com/a/28283340 ("Pandoc Markdown enables the 'escaped_line_breaks' extension by default")
* https://github.github.com/gfm/#hard-line-break
* https://spec.commonmark.org/0.30/#hard-line-breaks

**Comment by MikeTaylor** [comment]
Thanks for finding this -- it was new to me! It certainly works in PanDoc (with both HTML and PDF outputs) and in whatever random show-Markdown-as-HTML extension I have enabled in Chrome. If we are able to determine experimentally that it also works in GitHub's display of Markdown (which in practice is probably by far the most important use-case) we can make this our standard practice.

**Comment by dcrossleyau** [comment]
Yes it does work with GitHub Flavored Markdown. See the spec links in my comment.

I did make two demos  to verify display via GitHub::
* My PR comment above has one for the line-break preceding "rather than using".
* [The source](https://github.com/folio-org/folio-org.github.io/blob/48295900c3adb42efca7cce368e2e62c2620ef04/reference/api/index.md#versioned-api-docs) of my "versioned-api-docs" demo.

**Comment by MikeTaylor** [comment]
Excellent! Then that is what I will use going forward (assuming I remember). I think it's a terrible, terrible convention for a line-break; but still much less terrible than two trailing spaces!

**Comment by dcrossleyau** [comment]
I followed through to add FAQ: https://dev.folio.org/faqs/how-to-use-editorconfig/

**Comment by MikeTaylor** [comment]
Nice, thank you!

**Pull request merged**